google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/AddCloudLibrariesAction.java

```diff
@@ -35,6 +39,23 @@ public AddCloudLibrariesAction() {
         GoogleCloudToolsIcons.CLOUD);
   }
 
+  @Override
+  public void update(AnActionEvent e) {
+    boolean enableAddLibraries = false;
```

Comment:
might be cleaner over all if you made it more functional:

```
...findAny().map( //set visible true).orElseGet( //set visible to false);
```

Up to you if you think its cleaner

Reply:
or better yet: 

```
setVisible(..anyMatch(module -> isMavenModudel(module)));
```

Reply:
Interesting exercise, not sure if it's more readable, but looks neat :)

Reply:
might be even more readable if you take out the project optional stuff from that and just have 

```
if (e.getProject() == null) {
  setVisible(false);
  return;
}

// functional craziness
```

Reply:
That's what I wanted initially and then I thought "sooner or later someone will definitely want *everything* wrapped in one functional style run if it's already there".

Reply:
lol. i was really fine with what you had originally. I just like to make things functional for fun. Go with whichever you prefer.

Reply:
I think let's leave it functional here - I strongly suspect we'll get rid of this in a few days or a week anyway once we have Gradle/native message/wording added into libraries UI. WDYT?